mysql/mydecimal.go

```diff
@@ -1173,12 +1153,19 @@ func (d *MyDecimal) ToBin(precision, frac int) ([]byte, int) {
 		}
 	}
 	bin[0] ^= 0x80
-	return bin, errCode
+	return result, err
 }
 
 // FromBin Restores decimal from its binary fixed-length representation.
-func (d *MyDecimal) FromBin(bin []byte, precision, frac int) int {
-	var errcode = eDecOK
+func (d *MyDecimal) FromBin(bin []byte) error {
+	if len(bin) < 3 {
+		*d = zeroMyDecimal
+		return ErrBadNumber
+	}
+	precision := int(bin[0])
```

Comment:
Use 'int' during calculation, cast to 'int8' when set the result value, this way can avoid overflow issue.